models/src/main/protobuf/CasperMessage.proto

```diff
@@ -22,14 +22,19 @@ option (scalapb.options) = {
   preamble: "sealed trait CasperMessage"
 };
 
+message HasBlockRequest {
+  option (scalapb.message).extends = "CasperMessage";
+  bytes  hash       = 1;
+}
+
+message HasBlock {
+  option (scalapb.message).extends = "CasperMessage";
+  bytes  hash       = 1;
+}
 
 message BlockRequest {
   option (scalapb.message).extends = "CasperMessage";
-  //Fields are redundant to allow for validity check.
-  //Such a check can definitively distinguish this
-  //from other message types with similar serializations.
-  string base16Hash = 1;
-  bytes  hash       = 2;
+  bytes  hash       = 1;
 }
```

Comment:
You change the protocol.
Right now the integration test would simulate the way how a rnode request a block and send a block. So if you change the protocol, the test would fail.
According to the log https://drone.rchain-dev.tk/rchain/rchain/4641/7 line 1230, the integration test can not request the block right because the `hash` is messing around with the `base16Hash`.

But right now, the integration test is using protocol which stays in  [pyrchain](https://github.com/rchain/pyrchain). I need to updated the lib first. So I RECOMMEND you to mark the test xfail first. I create the ticket for that to fix it.

Reply:
https://rchain.atlassian.net/browse/RCHAIN-3663 ticket for that.

Reply:
Thx @zsluedem